app/DoctrineMigrations/Version20200416135127.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Surfnet\StepupMiddleware\Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+class Version20200416135127 extends AbstractMigration
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('CREATE VIEW `view_ra_candidate` AS
+(
+SELECT DISTINCT i.id          as identity_id,
+                i.institution,
+                i.common_name,
+                i.email,
+                i.name_id,
+                a.institution AS ra_institution
+FROM identity i
+         INNER JOIN vetted_second_factor vsf on vsf.identity_id = i.id
+         INNER JOIN institution_authorization a ON (
+        a.institution_role = \'select_raa\' AND a.institution_relation = i.institution
+    )
+WHERE NOT EXISTS(SELECT 1 FROM ra_listing AS l WHERE l.identity_id = i.id AND l.ra_institution = a.institution)
+);');
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function down(Schema $schema)
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('DROP VIEW `view_ra_candidate` ');
+        $this->addSql('DROP TABLE ra_candidate');
```

Comment:
I'm confused why the ra_candidate table is also dropped when the migration is downed. Would expect it to be dropped on the UP method. Could you explain this maybe also in an in-line comment?

Reply:
Thanks, that indeed shouldn't happen at all.

Reply:
Removed it completely. I'll consult our PM what should happen with it. Maybe we could better clean it the next release.

Reply:
The RA candidate could be removed so I added the drop statement in the `up` method and added a irreversible exception to the `down` method.